core/src/test/java/org/elasticsearch/index/mapper/core/DateFieldTypeTests.java

```diff
@@ -54,4 +67,49 @@ public void modify(MappedFieldType ft) {
             }
         });
     }
+
+    public void testIsFieldWithinQueryEmptyReader() throws IOException {
+        IndexReader reader = new MultiReader();
+        DateFieldType ft = new DateFieldType();
+        ft.setName("my_date");
+        assertEquals(Relation.DISJOINT, ft.isFieldWithinQuery(reader, "2015-10-12", "2016-04-03",
+                randomBoolean(), randomBoolean(), null, null));
+    }
+
+    public void testIsFieldWithinQuery() throws IOException {
+        Directory dir = newDirectory();
+        IndexWriter w = new IndexWriter(dir, new IndexWriterConfig(null));
+        long instant1 = DateFieldMapper.Defaults.DATE_TIME_FORMATTER.parser().parseDateTime("2015-10-12").getMillis();
+        long instant2 = DateFieldMapper.Defaults.DATE_TIME_FORMATTER.parser().parseDateTime("2016-04-03").getMillis();
+        Document doc = new Document();
+        LegacyLongField field = new LegacyLongField("my_date", instant1, Store.NO);
+        doc.add(field);
+        w.addDocument(doc);
+        field.setLongValue(instant2);
+        w.addDocument(doc);
+        DirectoryReader reader = DirectoryReader.open(w);
+        DateFieldType ft = new DateFieldType();
+        ft.setName("my_date");
+        assertEquals(Relation.INTERSECTS, ft.isFieldWithinQuery(reader, "2015-10-09", "2016-01-02",
+                randomBoolean(), randomBoolean(), null, null));
+        assertEquals(Relation.INTERSECTS, ft.isFieldWithinQuery(reader, "2016-01-02", "2016-06-20",
+                randomBoolean(), randomBoolean(), null, null));
+        assertEquals(Relation.INTERSECTS, ft.isFieldWithinQuery(reader, "2016-01-02", "2016-02-12",
+                randomBoolean(), randomBoolean(), null, null));
+        assertEquals(Relation.DISJOINT, ft.isFieldWithinQuery(reader, "2014-01-02", "2015-02-12",
+                randomBoolean(), randomBoolean(), null, null));
+        assertEquals(Relation.DISJOINT, ft.isFieldWithinQuery(reader, "2016-05-11", "2016-08-30",
+                randomBoolean(), randomBoolean(), null, null));
+        assertEquals(Relation.WITHIN, ft.isFieldWithinQuery(reader, "2015-09-25", "2016-05-29",
+                randomBoolean(), randomBoolean(), null, null));
+        assertEquals(Relation.WITHIN, ft.isFieldWithinQuery(reader, "2015-10-12", "2016-04-03",
+                true, true, null, null));
+        assertEquals(Relation.INTERSECTS, ft.isFieldWithinQuery(reader, "2015-10-12", "2016-04-03",
+                false, false, null, null));
+        assertEquals(Relation.INTERSECTS, ft.isFieldWithinQuery(reader, "2015-10-12", "2016-04-03",
+                false, true, null, null));
+        assertEquals(Relation.INTERSECTS, ft.isFieldWithinQuery(reader, "2015-10-12", "2016-04-03",
+                true, false, null, null));
+        IOUtils.close(reader, w, dir);
+    }
```

Comment:
Should we add tests which provide a custom timezone and a custom date math parser (custom date format)?